Code/Tools/ProjectManager/Source/GemCatalog/GemModel.cpp

```diff
@@ -496,6 +525,41 @@ namespace O3DE::ProjectManager
         return modelIndex.data(RoleNewVersion).toString();
     }
 
+    QString GemModel::GetMostCompatibleVersion(const QModelIndex& modelIndex)
+    {
+        const auto& versionList = modelIndex.data(RoleGemInfoVersions).value<QList<QVariant>>();
+        if (versionList.isEmpty())
+        {
+            return {};
+        }
+
+        const GemInfo& gemInfo = versionList.at(0).value<GemInfo>();
+        if (gemInfo.IsCompatible())
+        {
+            // most common case is this gem is an engine gem,
+            // or the gem is compatible
+            return gemInfo.m_version;
+        }
+
+        // versions are sorted from highest to lowest so return the first compatible version
+        for (const auto& versionVariant : versionList)
+        {
+            const GemInfo& variantGemInfo = versionVariant.value<GemInfo>();
+            if(variantGemInfo.IsCompatible())
+            {
+                return variantGemInfo.m_version;
+            }
+        }
```

Comment:
The for loop does the same thing as the if statement.
It checks in order that the gemInfo version is compatible from highest version to lowest version.

Therefore the if statement shouldn't be needed.